examples/custom-mailets/src/test/java/org/apache/james/examples/custom/listeners/BigMessageListenerTest.java

```diff
@@ -0,0 +1,169 @@
+        mailboxSession = MailboxSessionUtil.create(USER);
+        inboxId = mailboxManager.createMailbox(MailboxPath.forUser(USER, DefaultMailboxes.INBOX), mailboxSession).get();
+        inboxMessageManager = mailboxManager.getMailbox(inboxId, mailboxSession);
+
+        testee = new BigMessageListener(mailboxManager, messageIdManager);
+
+        resources.getEventBus().register(testee);
+    }
+
+    @Test
+    void listeningEventShouldNotAddFlagWhenSmallMessages() throws Exception {
+        ComposedMessageId composedId = inboxMessageManager.appendMessage(
+            MessageManager.AppendCommand.builder()
+                .build(smallMessage()),
+            mailboxSession);
+
+        assertThat(getMessageFlags(composedId.getMessageId()))
+            .allSatisfy(flags -> assertThat(flags.contains(BIG_MESSAGE)).isFalse());
+    }
+
+    @Test
+    void listeningEventShouldNotRemoveOtherFlagsWhenSmallMessages() throws Exception {
+        Flags appendMessageFlag = new Flags();
+        appendMessageFlag.add(Flags.Flag.SEEN);
+        appendMessageFlag.add(Flags.Flag.DRAFT);
+
+        ComposedMessageId composedId = inboxMessageManager.appendMessage(
+            MessageManager.AppendCommand.builder()
+                .withFlags(appendMessageFlag)
+                .build(smallMessage()),
+            mailboxSession);
+
+        assertThat(getMessageFlags(composedId.getMessageId()))
+            .allSatisfy(flags -> {
+                assertThat(flags.contains(Flags.Flag.SEEN)).isTrue();
+                assertThat(flags.contains(Flags.Flag.DRAFT)).isTrue();
+            });
+    }
+
+    @Test
+    void listeningEventShouldAddFlagWhenBigMessages() throws Exception {
+        ComposedMessageId composedId = inboxMessageManager.appendMessage(
+            MessageManager.AppendCommand.builder()
+                .build(bigMessage()),
+            mailboxSession);
+
+        assertThat(getMessageFlags(composedId.getMessageId()))
+            .allSatisfy(flags -> assertThat(flags.contains(BIG_MESSAGE)).isTrue());
+    }
+
+    @Test
+    void listeningEventShouldNotRemoveOtherFlagsWhenBigMessages() throws Exception {
+        Flags appendMessageFlag = new Flags();
+        appendMessageFlag.add(Flags.Flag.SEEN);
+        appendMessageFlag.add(Flags.Flag.DRAFT);
+
+        ComposedMessageId composedId = inboxMessageManager.appendMessage(
+            MessageManager.AppendCommand.builder()
+                .withFlags(appendMessageFlag)
+                .build(bigMessage()),
+            mailboxSession);
+
+        assertThat(getMessageFlags(composedId.getMessageId()))
+            .allSatisfy(flags -> {
+                assertThat(flags.contains(Flags.Flag.SEEN)).isTrue();
+                assertThat(flags.contains(Flags.Flag.DRAFT)).isTrue();
+                assertThat(flags.contains(BIG_MESSAGE)).isTrue();
+            });
+    }
+
+    @Test
+    void listeningEventShouldKeepBigMessageFlagWhenAlreadySet() throws Exception {
```

Comment:
missing a case of `when message size is equal to big message size`. But it's kinda hard to setup the scenario even with mockito. Maybe writing a simple test at unit test level + mocking is okay